torch/distributions/kl.py

```diff
@@ -121,10 +128,211 @@ def kl_divergence(p, q):
+    if p.support.lower_bound >= q.support.lower_bound:
+        #  From http://www.mast.queensu.ca/~communications/Papers/gil-msc11.pdf
+        scale_ratio = p.scale / q.scale
+        alpha_ratio = q.alpha / p.alpha
+        t1 = q.alpha * scale_ratio.log()
+        t2 = -alpha_ratio.log()
+        return t1 + t2 + alpha_ratio - 1
+    else:
+        return torch.new(p.scale.size()).fill_(float('inf'))
+
+@register_kl(Uniform, Uniform)
+def _kl_uniform_uniform(p, q):
+    if q.support.lower_bound <= p.support.lower_bound and q.support.upper_bound >= p.support.upper_bound:
+        return ((q.high - q.low) / (p.high - p.low)).log()
+    else:
+        return torch.new(q.high.size()).fill_(float('inf'))
+
+# Different distributions
+
+@register_kl(Beta, Exponential)
+def _kl_beta_exponential(p, q):
+    return -p.entropy() - q.rate.log() + q.rate * (p.alpha / (p.alpha + p.beta))
+
+@register_kl(Beta, Gamma)
+def _kl_beta_gamma(p, q):
+    t1 = -p.entropy()
+    t2 = q.alpha.lgamma() - q.alpha * q.beta.log()
+    t3 = (q.alpha - 1) * (p.alpha.digamma() - (p.alpha + p.beta).digamma())
+    t4 = q.beta * p.alpha / (p.alpha + p.beta)
+    return t1 + t2 - t3 + t4
+
+@register_kl(Beta, Normal)
+def _kl_beta_normal(p, q):
+    E_beta = p.alpha / (p.alpha + p.beta)
+    var_normal = q.std.pow(2)
+    t1 = -p.entropy()
+    t2 = 0.5 * (var_normal * 2 * math.pi).log()
+    t3 = (E_beta * (1 - E_beta) / (p.alpha + p.beta + 1) + E_beta.pow(2)) * 0.5
+    t4 = q.mean * E_beta
+    t5 = q.mean.pow(2) * 0.5
+    return t1 + t2 + (t3 - t4 + t5) / var_normal
+
+@register_kl(Beta, Pareto)
+def _kl_beta_pareto(p, q):
+    return torch.new(p.rate.size()).fill_(float('inf'))
+
+@register_kl(Beta, Uniform)
+def _kl_beta_uniform(p, q):
+    if q.support.lower_bound <= p.support.lower_bound and q.support.upper_bound >= p.support.upper_bound:
+        return -p.entropy() + (q.high - q.low).log()
+    else:
+        return torch.new(p.rate.size()).fill_(float('inf'))
+
+@register_kl(Exponential, Beta)
+def _kl_exponential_beta(p, q):
+    return torch.new(p.rate.size()).fill_(float('inf'))
+
+@register_kl(Exponential, Gamma)
+def _kl_exponential_gamma(p, q):
+    t1 = q.alpha * q.beta.log() - torch.lgamma(q.alpha)
+    t2 = (q.alpha - 1) * (p.rate.log() + 0.57721566490153286060) / p.rate
+    t3 = q.beta / p.rate
+    return t1 - t2 - t3
+
+@register_kl(Exponential, Normal)
+def _kl_exponential_normal(p, q):
+    var_normal = q.std.pow(2)
+    rate_sqr = p.rate.pow(2)
+    t1 = 0.5 * torch.log(rate_sqr * var_normal * 2 * math.pi)
+    t2 = rate_sqr.reciprocal()
+    t3 = q.mean / p.rate
+    t4 = q.mean.pow(2) * 0.5
+    return t1 - 1 + (t2 - t3 + t4) / var_normal
+
+@register_kl(Exponential, Pareto)
+def _kl_exponential_pareto(p, q):
+    return torch.new(p.rate.size()).fill_(float('inf'))
+
+@register_kl(Exponential, Uniform)
+def _kl_exponential_uniform(p, q):
+    return torch.new(p.rate.size()).fill_(float('inf'))
+
+@register_kl(Gamma, Beta):
```

Comment:
Minor nit over here. I have changed it locally. Will push with more changes soon.